src/modules/NodeCollection.js

```diff
@@ -0,0 +1,133 @@
+export class NodeCollection {
+  constructor (nodes) {
+    this.nodes = nodes
+  }
+
+  _isUndefined (value) {
+    return value === undefined || value === null || value === ''
+  }
+
+  _strValue (str, caseSensitive) {
+    return caseSensitive ? `${str}` : `${str}`.toLowerCase()
+  }
+
+  _createStringFilter (filterValue, caseSensitive) {
+    if (this._isUndefined(filterValue)) {
+      filterValue = ''
+    }
+    const strFilter = this._strValue(filterValue, caseSensitive)
+    return value => this._strValue(value, caseSensitive).indexOf(strFilter) >= 0
+  }
+
+  _createRegexFilter (filterValue, caseSensitive) {
+    if (this._isUndefined(filterValue)) {
+      filterValue = ''
+    }
+    const strFilter = this._strValue(filterValue, caseSensitive)
+    return value => !!this._strValue(value, caseSensitive).match(strFilter)
```

Comment:
You should check that `strFilter` is a valid regex. This could throw an error and break the ui.

To reproduce the issue try to insert `*` in the regex, all filters will stop working

Reply:
Done.